frontend/catalyst/compiler.py

```diff
@@ -107,233 +104,115 @@ def get_lib_path(project, env_var):
+    (
+        "QuantumCompilationPass",
+        [
+            "lower-gradients",
+            "adjoint-lowering",
+            "convert-arraylist-to-memref",
+        ],
+    ),
+    (
+        "BufferizationPass",
+        [
+            "one-shot-bufferize{dialect-filter=memref}",
+            "inline",
+            "gradient-bufferize",
+            "scf-bufferize",
+            "convert-tensor-to-linalg",  # tensor.pad
+            "convert-elementwise-to-linalg",  # Must be run before --arith-bufferize
+            "arith-bufferize",
+            "empty-tensor-to-alloc-tensor",
+            "func.func(bufferization-bufferize)",
+            "func.func(tensor-bufferize)",
+            "func.func(linalg-bufferize)",
+            "func.func(tensor-bufferize)",
+            "quantum-bufferize",
+            "func-bufferize",
+            "func.func(finalizing-bufferize)",
+            # "func.func(buffer-hoisting)",
+            "func.func(buffer-loop-hoisting)",
+            # "func.func(buffer-deallocation)",
+            "convert-bufferization-to-memref",
+            "canonicalize",
+            # "cse",
+            "cp-global-memref",
+        ],
+    ),
+    (
+        "MLIRToLLVMDialect",
+        [
+            "func.func(convert-linalg-to-loops)",
+            "convert-scf-to-cf",
+            # This pass expands memref operations that modify the metadata of a memref (sizes, offsets,
+            # strides) into a sequence of easier to analyze constructs. In particular, this pass
+            # transforms operations into explicit sequence of operations that model the effect of this
+            # operation on the different metadata. This pass uses affine constructs to materialize
+            # these effects. Concretely, expanded-strided-metadata is used to decompose memref.subview
+            # as it has no lowering in -finalize-memref-to-llvm.
+            "expand-strided-metadata",
+            "lower-affine",
+            "arith-expand",  # some arith ops (ceildivsi) require expansion to be lowered to llvm
+            "convert-complex-to-standard",  # added for complex.exp lowering
+            "convert-complex-to-llvm",
+            "convert-math-to-llvm",
+            # Run after -convert-math-to-llvm as it marks math::powf illegal without converting it.
+            "convert-math-to-libm",
+            "convert-arith-to-llvm",
+            "finalize-memref-to-llvm{use-generic-functions}",
+            "convert-index-to-llvm",
+            "convert-gradient-to-llvm",
+            "convert-quantum-to-llvm",
+            "emit-catalyst-py-interface",
+            # Remove any dead casts as the final pass expects to remove all existing casts,
+            # but only those that form a loop back to the original type.
+            "canonicalize",
+            "reconcile-unrealized-casts",
+        ],
+    ),
+]
+
+# FIXME: Figure out how to encode Enzyme pipeline. Probably we should make it the same way we make
+# CppCompiler
```

Comment:
Agree. I am removing this code now, but let me know if you think that we need to preserve it as a comment or fix this knowledge in some other form.

Reply:
I think this needs to be looked at a bit more closely before merging to main. The gradient lowering still introduces the enzyme symbols, but because enzyme is never triggered, the symbols are undefined.

```
E           OSError: /tmp/tmpdpgogbl0/compiled.so: undefined symbol: __enzyme_autodiff0

../../.pyenv/versions/3.10.11/lib/python3.10/ctypes/__init__.py:374: OSError
```

Reply:
I can confirm test failures with these messages.